File: ui/src/app/core/dtos/leaveRequests.dto.ts
export interface LeaveRequestsDto {
  leaveRequestId?: number;

  userId?: number;
  leaveTypeId?: number;

  startedDate?: string | Date;
  finishDate?: string | Date;

  teamLeaderId?: number;
  teamLeaderApprovalDate?: string | Date;

  hrSpecialistId?: number;
  hrSpecialistApprovalDate?: string | Date;

  seniorManagerId?: number;
  seniorManagerApprovalDate?: string | Date;

  rejectionStatement?: string;

  rejectingUserId?: number;

  dayCount?: number;

  isActive?: boolean;
}

File: api/Dtos/LeaveRequestsDto.cs
namespace api.Dtos
{
    public class LeaveRequestsDto
    {
        public int LeaveRequestId { get; set; }

        public int UserId { get; set; }
        public int LeaveTypeId { get; set; }
        public string LeaveTypeName { get; set; } = string.Empty;

        public DateTime StartedDate { get; set; }
        public DateTime FinishDate { get; set; }

        public int DayCount { get; set; }

        public bool IsActive { get; set; }
    }
}